### ISSUE TITLE
reportTypedDictNotRequiredAccess inconsistent behavior depending on how the TypedDict was instantiated

### ISSUE THREAD

**Issue by ITProKyle** [microsoft/pyright#2183]
**Describe the bug**

Given the following code, `reportTypedDictNotRequiredAccess` is raised when trying to set the value of a `TypedDict` key that is not required. However, this only happens for a variable that has been annotated with the `TypedDict`. The error is not raised when the `TypedDict` is used directly.

```python
"""Output:

example.py:19:1 - error: Could not access item in TypedDict
    "key1" is not a required key in "MyData", so access may result in runtime exception (reportTypedDictNotRequiredAccess)

"""
from typing import TypedDict


class MyData(TypedDict, total=False):
    key0: str
    key1: str


example_dict_from_typed_dict = MyData(key0="foo")
example_dict_from_typed_dict["key1"] = "bar"

example_dict_from_type_hint: MyData = {"key0": "foo"}
example_dict_from_type_hint["key1"] = "bar"
```

**To Reproduce**

1. Copy the above code into a file name `example.py`.
2. Using `pyright@1.1.161`, run `npx pyright ./example.py`.

**Expected behavior**

I would expect both usages to not produce an error when type-checking.

**VS Code extension or command-line**

command-line

**Additional context**

```shell
$ python --version
Python 3.9.6
```
```shell
$ npm --version
7.20.0
```

This code does not result in an error when using `pyright@1.1.160`.

**Comment by erictraut** [comment]
Thanks for the bug report. Yes, this is definitely a bug. This error message should never appear for _set_ (write) accesses, only _get_ (read) operations.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.162, which I just published. It will also be in the next release of pylance.
